### PR TITLE
[12.0][FIX] sentry: Sentry sdk removed text_type from _compat utils

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: setup/sentry/setup.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
     odoo_addon={
         'external_dependencies_override': {
             'python': {
-                'sentry_sdk': 'sentry_sdk>=1.17.0',
+                'sentry_sdk': 'sentry_sdk>=1.17.0,<=1.45.0',
```
Comment (sbidoul) [inline setup/sentry/setup.py]:
```suggestion
                'sentry_sdk': 'sentry_sdk>=1.17.0,<2',
```

**Review by moylop260 — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
OCB job failed on 
```
2024-09-16T16:35:07.4244950Z 2024-09-16 16:35:07,423 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: ERROR: test_dns_challenge_error_on_missing_provider (odoo.addons.letsencrypt.tests.test_letsencrypt.TestLetsencrypt) 
2024-09-16T16:35:07.4322735Z 2024-09-16 16:35:07,431 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: Traceback (most recent call last): 
2024-09-16T16:35:07.4398003Z 2024-09-16 16:35:07,439 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/__w/server-tools/server-tools/letsencrypt/tests/test_letsencrypt.py", line 135, in test_dns_challenge_error_on_missing_provider 
2024-09-16T16:35:07.4472257Z 2024-09-16 16:35:07,446 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     self.env['letsencrypt']._cron() 
2024-09-16T16:35:07.4550984Z 2024-09-16 16:35:07,454 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/__w/server-tools/server-tools/letsencrypt/models/letsencrypt.py", line 232, in _cron 
2024-09-16T16:35:07.4633167Z 2024-09-16 16:35:07,462 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     authzr = client.new_order(csr) 
2024-09-16T16:35:07.4713823Z 2024-09-16 16:35:07,470 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/opt/odoo-venv/lib/python3.6/site-packages/acme/client.py", line 711, in new_order 
2024-09-16T16:35:07.4794054Z 2024-09-16 16:35:07,478 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     authorizations.append(self._authzr_from_response(self._post_as_get(url), uri=url)) 
2024-09-16T16:35:07.4876673Z 2024-09-16 16:35:07,487 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/opt/odoo-venv/lib/python3.6/site-packages/acme/client.py", line 837, in _post_as_get 
2024-09-16T16:35:07.4960418Z 2024-09-16 16:35:07,495 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     return self._post(*new_args, **kwargs) 
2024-09-16T16:35:07.5048538Z 2024-09-16 16:35:07,504 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/opt/odoo-venv/lib/python3.6/site-packages/acme/client.py", line 103, in _post 
2024-09-16T16:35:07.5134274Z 2024-09-16 16:35:07,512 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     return self.net.post(*args, **kwargs) 
2024-09-16T16:35:07.5220951Z 2024-09-16 16:35:07,521 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/opt/odoo-venv/lib/python3.6/site-packages/acme/client.py", line 1272, in post 
2024-09-16T16:35:07.5309524Z 2024-09-16 16:35:07,530 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     return self._post_once(*args, **kwargs) 
2024-09-16T16:35:07.5401574Z 2024-09-16 16:35:07,539 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/opt/odoo-venv/lib/python3.6/site-packages/acme/client.py", line 1286, in _post_once 
2024-09-16T16:35:07.5490960Z 2024-09-16 16:35:07,548 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     response = self._check_response(response, content_type=content_type) 
2024-09-16T16:35:07.5582902Z 2024-09-16 16:35:07,557 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `   File "/opt/odoo-venv/lib/python3.6/site-packages/acme/client.py", line 1131, in _check_response 
2024-09-16T16:35:07.5676411Z 2024-09-16 16:35:07,567 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: `     raise messages.Error.from_json(jobj) 
2024-09-16T16:35:07.5787032Z 2024-09-16 16:35:07,576 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: ` acme.messages.Error: urn:ietf:params:acme:error:malformed :: The request message was malformed :: No such authorization 
2024-09-16T16:35:07.5875397Z 2024-09-16 16:35:07,587 1163 [1;32m[1;49mINFO[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: Ran 16 tests in 11.241s 
2024-09-16T16:35:07.5877376Z 2024-09-16 16:35:07,587 1163 [1;31m[1;49mERROR[0m odoo odoo.addons.letsencrypt.tests.test_letsencrypt: FAILED 
```
Green on retry. Let's try again here as well

/ocabot merge patch

**Pull request merged**